switchdev_utils/config-service/switchdev_network_config

```diff
@@ -0,0 +1,548 @@
+#!/usr/bin/python
+
+
+import logging
+import os
+import subprocess
+import sys
+import time
+import yaml
+
+from collections import OrderedDict
+
+CONFIG_FILE = "/etc/switchdev-config/config.yaml"
+LOG_FILE = '/var/log/switchdev_network_config.log'
+
+SYS_CLASS_NET = '/sys/class/net'
+MLNX_VENDOR_ID = "0x15b3"
+CX4_DEVICE_ID = "0x1013"
+CX4LX_DEVICE_ID = "0x1015"
+VF_PCI_PATH = "/sys/bus/pci/devices/%(vf_pci)s/driver"
+MLNX_UNBIND_FILE_PATH = "/sys/bus/pci/drivers/mlx5_core/unbind"
+MLNX_BIND_FILE_PATH = "/sys/bus/pci/drivers/mlx5_core/bind"
+SRIOV_BIND_CONFIG_FILE = "/var/lib/sriov_bind_config.yaml"
+
+UDEV_RULE_FILE = '/etc/udev/rules.d/90-sriov-config.rules'
+COMMON_IFCFG_DATA = 'ONBOOT=yes\nHOTPLUG=no\nNM_CONTROLLED=' \
+                    'no\nPEERDNS=no\nBOOTPROTO=none\n'
+MAX_RETRIES = 10
+
+SRIOV_CONFIG_SERVICE_FILE = "/etc/systemd/system/sriov_config.service"
+SRIOV_CONFIG_SERVICE_CONTENT = """[Unit]
+Description=SR-IOV numvfs configuration
+After=systemd-udev-settle.service
+Before=openvswitch.service network.service
+[Service]
+Type=oneshot
+ExecStart=/etc/switchdev-config/switchdev_network_config sriov_config
+[Install]
+WantedBy=multi-user.target
+"""
+
+SRIOV_BIND_SERVICE_FILE = "/etc/systemd/system/sriov_bind.service"
+SRIOV_BIND_SERVICE_CONTENT = """[Unit]
+Description=SR-IOV vfs binding
+After=network.service sriov_config.service
+[Service]
+Type=oneshot
+ExecStart=/etc/switchdev-config/switchdev_network_config sriov_bind
+[Install]
+WantedBy=multi-user.target
+"""
+# Map of all net devices and their ifcfg files
+ifcfg_interfaces_map = OrderedDict()
+mlnx_vfs_pci_list = []
+
+logging.basicConfig(filename=LOG_FILE,
+                    level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def get_file_data(filename):
+    if not os.path.exists(filename):
+        logger.warning("File $s does not exist" % filename)
+        return ''
+    try:
+        with open(filename, 'r') as f:
+            return f.read().strip()
+    except IOError as e:
+        logger.error("Error reading file: %s\n%s" % (filename, str(e)))
+        return ''
+
+
+def get_config():
+    if os.path.exists(CONFIG_FILE):
+        config_data = get_file_data(CONFIG_FILE)
+        try:
+            config = yaml.safe_load(config_data)
+            if config:
+                return config
+            else:
+                logger.error("No configuration is provided.. Exiting")
+                exit(1)
+        except yaml.YAMLError as e:
+            logger.error("Error parsing config file: %s" % CONFIG_FILE)
+            exit(1)
+    else:
+        logger.error("Configuration file does not exist.. Exiting")
+        exit(1)
+
+
+def run_command(*cmd):
+    try:
+        p = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                             stderr=subprocess.PIPE)
+        out, err = p.communicate()
+        ret = p.returncode
+        if ret != 0:
+            err_msg = "Falied to execute %s \n%s" % (' '.join(cmd), err)
```

Comment:
not sure how nice the `\n` will look in log, why not just :  `Failed to execute: %s. %s`